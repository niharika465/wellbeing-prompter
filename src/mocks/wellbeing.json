[
  {
    "sku": "sku_101",
    "value": "terrible",
    "label": "Terrible",
    "color": "red"
  },
  {
    "sku": "sku_102",
    "value": "bad",
    "label": "Bad",
    "color":"orange"
  },
  {
    "sku": "sku_103",
    "value": "alright",
    "label": "Alright",
     "color":"purple"
  },
  {
    "sku": "sku_104",
    "value": "pretty_good",
    "label": "Pretty Good",
    "color":"blue"
  },
  {
    "sku": "sku_105",
    "value": "fantastic",
    "label": "Fantastic",
    "color": "green"
  }
]
