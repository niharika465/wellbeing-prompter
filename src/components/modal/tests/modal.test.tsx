import { render, screen, fireEvent } from '@testing-library/react';
import Modal from '../modal';
import wellbeing from '../../../mocks/wellbeing.json';

type ChoiceCardProps = {
  label: string;
  isSelected: boolean;
  handleClick: () => void;
};

jest.mock(
  '../../choice-card/choice-card',
  () =>
    ({ label, handleClick, isSelected }: ChoiceCardProps) => (
      <div
        data-testid="choice-card"
        className={isSelected ? 'selected' : ''}
        onClick={handleClick}
      >
        {label}
      </div>
    )
);

describe('Modal Component', () => {
  const handleClick = jest.fn();
  const setUserChoice = jest.fn();
  const userChoice = '';

  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('renders without crashing when closed', () => {
    const { asFragment } = render(
      <Modal
        isOpen={false}
        handleClick={handleClick}
        userChoice={userChoice}
        setUserChoice={setUserChoice}
      />
    );
    expect(asFragment()).toMatchSnapshot();
    const modal = screen.queryByText('Wellbeing Check-in');
    expect(modal).not.toBeInTheDocument(); // Modal should not be visible
  });

  test('renders correctly when open', () => {
    const { asFragment } = render(
      <Modal
        isOpen={true}
        handleClick={handleClick}
        userChoice={userChoice}
        setUserChoice={setUserChoice}
      />
    );
    expect(asFragment()).toMatchSnapshot();
    const modalHeader = screen.getByText('Wellbeing Check-in');
    expect(modalHeader).toBeInTheDocument(); // Modal should be visible
  });

  test('calls handleClick when back icon is clicked', () => {
    render(
      <Modal
        isOpen={true}
        handleClick={handleClick}
        userChoice={userChoice}
        setUserChoice={setUserChoice}
      />
    );

    const backIcon = screen.getByTestId('ArrowBackIcon');
    fireEvent.click(backIcon);

    expect(handleClick).toHaveBeenCalledTimes(1); // handleClick should be called
  });

  test('calls handleClick when close icon is clicked', () => {
    render(
      <Modal
        isOpen={true}
        handleClick={handleClick}
        userChoice={userChoice}
        setUserChoice={setUserChoice}
      />
    );

    const closeIcon = screen.getByTestId('CloseIcon');
    fireEvent.click(closeIcon);

    expect(handleClick).toHaveBeenCalledTimes(1); // handleClick should be called
  });

  test('renders choice cards and calls setUserChoice', () => {
    render(
      <Modal
        isOpen={true}
        handleClick={handleClick}
        userChoice={userChoice}
        setUserChoice={setUserChoice}
      />
    );

    wellbeing.forEach(({ label }) => {
      expect(screen.getByText(label)).toBeInTheDocument(); // Each choice label should be rendered
    });

    const choiceCard = screen.getByText(wellbeing[0].label);
    fireEvent.click(choiceCard);

    expect(setUserChoice).toHaveBeenCalledWith(wellbeing[0].value); // setUserChoice should be called with the correct value
  });

  test('CustomButton should be disabled if no userChoice', () => {
    render(
      <Modal
        isOpen={true}
        handleClick={handleClick}
        userChoice={userChoice}
        setUserChoice={setUserChoice}
      />
    );

    const button = screen.getByText('Continue');
    expect(button).toHaveClass('btn--disabled'); // Button should be disabled
  });

  test('CustomButton should not be disabled if userChoice is set', () => {
    const userChoice = wellbeing[0].value;

    render(
      <Modal
        isOpen={true}
        handleClick={handleClick}
        userChoice={userChoice}
        setUserChoice={setUserChoice}
      />
    );

    const button = screen.getByText('Continue');
    expect(button).not.toHaveClass('btn--disabled'); // Button should be enabled
  });
});
