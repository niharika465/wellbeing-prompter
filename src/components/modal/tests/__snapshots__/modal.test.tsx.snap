// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component renders correctly when open 1`] = `
<DocumentFragment>
  <div
    class="modal"
  >
    <div
      class="modal__content"
    >
      <div
        class="header"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-ysecw-MuiSvgIcon-root"
          data-testid="ArrowBackIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20z"
          />
        </svg>
        <div
          class="heading"
        >
          Wellbeing Check-in
        </div>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-ysecw-MuiSvgIcon-root"
          data-testid="CloseIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
      </div>
      <div
        class="choices"
      >
        <div
          class="choices__heading"
        >
          Hello! How are you feeling today?
        </div>
        <div
          class="choices__container"
        >
          <div
            class=""
            data-testid="choice-card"
          >
            Terrible
          </div>
          <div
            class=""
            data-testid="choice-card"
          >
            Bad
          </div>
          <div
            class=""
            data-testid="choice-card"
          >
            Alright
          </div>
          <div
            class=""
            data-testid="choice-card"
          >
            Pretty Good
          </div>
          <div
            class=""
            data-testid="choice-card"
          >
            Fantastic
          </div>
        </div>
      </div>
      <div
        class="btn btn--disabled"
      >
        Continue
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Modal Component renders without crashing when closed 1`] = `<DocumentFragment />`;
